crates/vault-registry/src/lib.rs

```diff
@@ -1061,18 +1081,18 @@ impl<T: Config> Pallet<T> {
         let mut new_vault = Self::get_rich_vault_from_id(&new_vault_id)?;
 
         if old_vault.data.is_liquidated() {
-            let old_vault_collateral = CurrencySource::<T>::Collateral(old_vault_id.clone());
-
-            // transfer old-vault's collateral to liquidation_vault
-            let to_be_transfered_collateral = Self::calculate_collateral(
-                old_vault_collateral.current_balance()?,
+            let to_be_transferred = Self::calculate_collateral(
+                old_vault.data.liquidated_collateral,
                 tokens,
                 old_vault.data.to_be_redeemed_tokens,
             )?;
+            old_vault.decrease_liquidated_collateral(to_be_transferred)?;
```

Comment:
Can we do this from within transfer_funds?

Reply:
Currently this is also used to transfer collateral on slashing, so would prefer to leave as is for now.